### DISCUSSION TITLE
Prefer "by MXID" login flow over others

### DISCUSSION THREAD

**Discussion by HarHarLinks** [element-hq/element-meta#99] [Ideas]
I have been looking at the login flow here https://github.com/vector-im/element-android/issues/4382#issuecomment-982651171

when I open element android, my only option is to "get started"
then it faces me with 3 options: matrix.org server, EMS server, Other. Oh wait there is fairly hidden another option "Sign in with Matrix ID".

with the login flow in image above, that becomes:
when I open element android, my options are to "get started" or "already have an account". both however take me to the same next screen (wtf?):
then it faces me with 3 options: matrix.org server, EMS server, Other. Oh wait there is fairly hidden another option "Sign in with Matrix ID". IMO, pressing "already have an account" is an obvious case of taking a user straight to "login with MXID".

users are used to give their login name (e.g. me@provider.tld for email) and a password. the 3 default options however are about configuring a server first, which is evidently mostly unneccessary since the hidden 4th option is exactly the expected login flow we're used to. it discovers our homeserver automatically since it's part of our MXID. hence it should be the primary option for logging in.
i don't know about how it looks on iOS
I do know that the same MXID only login flow is available on web/desktop, although it's **hidden**: it defaults to matrix.to, but ignore that and enter your full MXID and it auto-discovers your proper homeserver when you continue to the password field. so the situation on desktop is even worse, since the MXID flow isn't even hinted at at all. (it also advertises using matrix.org as "the biggest server" as if that was a good thing, and also something matrix.org said they want to move away from.)

why entering a homeserver separately (people are educated to not know what servers are, see discord "servers") be better (simpler) than a single "username" that contains the server info? an additional confusion is introduced with delegation: which of the following is/are correct?
- example.org
- https://example.org
- matrix.example.org
- https://matrix.example.org

ignoring other variants one might try e.g. plain http://, users should not be expected to know the exact subdomain of their server, which is only relevant "once": when logging in.

email clients have moved to the same scheme: if I enter me@myself.tld in my email client, it will try to look up or failing that guess the proper IMAP and SMTP server for myself.tld
also from email I'm already educated to a server part in my "username".

All of these are real issues I continue having to face every time one of my users needs to log in.

This seems a really obvious switch and much preferable for so many reasons to me. What is holding it back? Why is it instead demoted to a second class login flow by either looking different (android) or not visibly hinted at at all?

**Comment by kittykat** [comment]
@HarHarLinks what do you think of the new login flows? I believe they were merged after you started this discussion

**Comment by HarHarLinks** [comment]
I think the web login flow has not changed. I would love to test the android changes once it is updated to a more current version than 1.4.14 on F-Droid.
